src/replication/replication_manager.cpp

```diff
@@ -0,0 +1,240 @@
+  //   REPLICA NAME
+  //   REPLICA HOSTNAME
+  //   REPLICA PORT
+  // Repeated and separated by newlines.
+  std::ifstream hosts_file(replication_hosts_path);
+  if (!hosts_file.is_open()) {
+    REPLICATION_LOG_ERROR(fmt::format("Unable to open file: {}", replication_hosts_path));
+    return;
+  }
+  std::string line;
+  std::string replica_name;
+  std::string replica_hostname;
+  uint16_t replica_port;
+  for (int ctr = 0; std::getline(hosts_file, line); ctr = (ctr + 1) % 4) {
+    switch (ctr) {
+      case 0:
+        // Ignored line.
+        break;
+      case 1:
+        replica_name = line;
+        break;
+      case 2:
+        replica_hostname = line;
+        break;
+      case 3:
+        replica_port = std::stoi(line);
+        // All information parsed.
+        if (identity_ == replica_name) {
+          // For our specific identity, check that the port is right.
+          NOISEPAGE_ASSERT(replica_port == port_, "Mismatch of identity/port combo in replica config.");
+        } else {
+          // Connect to the replica.
+          ReplicaConnect(replica_name, replica_hostname, replica_port);
+        }
+        break;
+      default:
+        NOISEPAGE_ASSERT(false, "Impossible.");
+        break;
+    }
+  }
+  hosts_file.close();
+}
+
+void ReplicationManager::ReplicaConnect(const std::string &replica_name, const std::string &hostname, uint16_t port) {
+  replicas_.try_emplace(replica_name, messenger_, replica_name, hostname, port);
+}
+
+void ReplicationManager::ReplicaSend(const std::string &replica_name, const ReplicationManager::MessageType type,
+                                     const std::string &msg, bool block) {
+  std::unique_lock<std::mutex> lock(mutex_);
+  REPLICATION_LOG_INFO(fmt::format("Sending to replication message to: {} with type {}", replica_name, type));
+  bool completed = false;
+  auto callback = [this, &completed](common::ManagedPointer<messenger::Messenger> messenger,
+                                                      std::string_view sender_id, std::string_view message,
+                                                      uint64_t recv_cb_id) {
+    completed = true;
+    cvar_.notify_all();
+  };
+  messenger_->SendMessage(GetReplicaConnection(replica_name), msg, callback, static_cast<uint64_t>(type));
+
+  if (block) {
+    // If the caller requested to block until the operation was completed, the thread waits.
+    cvar_.wait(lock, [&completed] { return completed; });
+  }
+  
+  lock.unlock();
+}
+
+void ReplicationManager::ReplicaAck(const std::string &replica_name) {
+  std::unique_lock<std::mutex> lock(mutex_);
+  REPLICATION_LOG_INFO("Sending ack to: " + replica_name);
+  messenger_->SendMessage(GetReplicaConnection(replica_name), ack_message_identifier_, nullptr,
+                          static_cast<uint64_t>(ReplicationManager::MessageType::ACK));
+  lock.unlock();
+}
+
+void ReplicationManager::EventLoop(common::ManagedPointer<noisepage::messenger::Messenger> messenger,
```

Comment:
I am not able to get the receiver ids in this case, only the sender ids, so I can't find a way to just use the defined enums for splitting cases. Is there another way for me to do that? Otherwise, I would just do something similar to this. (Using the enums instead of those temporary strings).